Reading List/Reading List/Controllers/BookController.swift

```diff
@@ -0,0 +1,145 @@
+*/
+
+import Foundation
+
+class BookController {
+    
+    var books: [Book] = []
+    
+    var readingListURL: URL? {
+        let fileManager = FileManager.default
+        guard let documentsDirectory = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            return nil
+        }
+        // Users/Angel/Documents
+        let finalLocation = documentsDirectory.appendingPathComponent("ReadingList.plist")
+        return finalLocation
+    }
+    
+    
+    func saveToPersistentStore(){
+        guard let url = readingListURL else { return } // ensuring there is a valid save destination
+        
+        let encoder = PropertyListEncoder()
+        do{
+            let data = try encoder.encode(books)
+            try data.write(to: url)
+        } catch{
+            print(error)
+        }
+    }
+    
+    func loadFromPersistentStore(){
+        
+        do{
+            guard let url = readingListURL else { return }
+            
+            let data = try Data(contentsOf: url)
+            let decoder = PropertyListDecoder()
+            let decodedBooks = try decoder.decode([Book].self, from: data)
+            books = decodedBooks
+        } catch {
+            print ("no url or no data at url")}
+    }
+    
+    func Create(title: String, reason : String){
+        let newBook = Book(title: title, reasonToRead: reason, hasBeenRead: false)
+        books.append(newBook)
+        saveToPersistentStore()
+    }
+    
+    func Delete(bookToDelete : Book){
+        guard let index = books.index(of: bookToDelete) else { return }
+                books.remove(at: index)
+                saveToPersistentStore()
+    }
+    
+    func updateHasBeenRead(for book: Book){
+        guard let index = books.index(of: book) else { return }
+                let oldBook = books[index]
+                let oldBookTitle = oldBook.title
+                let oldBookReason = oldBook.reasonToRead
+                let oldBookRead = oldBook.hasBeenRead
+                books.remove(at: index)
+                let newBook = Book(title: oldBookTitle, reasonToRead: oldBookReason, hasBeenRead: !oldBookRead)
+                books.insert(newBook, at: index)
+            saveToPersistentStore()
+    }
+    
+    
+    func updateTitleOrReason(for book: Book, newTitle title: String?, newReason reason: String?){
+        guard let index = books.index(of: book) else { return }
+                let oldBookTitle = books[index].title
+                let oldBookReason = books[index].reasonToRead
+                let oldBookRead = books[index].hasBeenRead
+                books.remove(at: index)
+                if title != nil && reason != nil {
+                    let updatedBook = Book(title: title!, reasonToRead: reason!, hasBeenRead: oldBookRead)
+                    books.insert(updatedBook, at: index)
```

Comment:
You should check `title` and `reason` in `BookDetailViewController` when user wants to save it. If those values are empty then do not run `updateTitleOrReason`. And later you will learn and present an alert to inform the user